injector/src/main/java/edu/ucr/cs/riple/injector/Injector.java

```diff
@@ -171,6 +172,22 @@ public static CompilationUnit parse(
     StaticJavaParser.setConfiguration(parserConfiguration);
     try {
       return StaticJavaParser.parse(path);
+    } catch (ParseProblemException e) {
+      // The original exception is not useful for the user. We should provide a more informative one
+      String message = e.getMessage();
+      // If the message contains the stack trace, we should remove it.
+      int index = message.indexOf("Problem stacktrace :");
+      message = index == -1 ? message : message.substring(0, index);
+      System.err.println(
+          "javaparser was not able to parse file at: "
+              + path
+              + "\n"
+              + message
+              + "\n"
+              + "Shutting down.");
+      // Exit with error code 1 to indicate failure.
+      System.exit(1);
+      return null;
```

Comment:
Rather than printing and calling `System.exit` here, why not create a wrapper exception object with the desired message and rethrow that?  It just seems like this is not the right place to decide if this is a fatal error that should exit the JVM.  Plus, directly calling `System.exit` makes it very difficult to write a test for this scenario.

Reply:
Sure, that sounds better. 7e4dfb13e2c74a75509d830638bf0fc54cb31ea9